services/server/src/server/apiv1/verification/private/stateless/private.stateless.handlers.ts

```diff
@@ -0,0 +1,406 @@
+      // Override so that it doesn't generate auxdata positions
+    };
+
+    Object.defineProperty(compilation, "creationBytecodeCborAuxdata", {
+      value: creationCodeArtifacts.cborAuxdata,
+    });
+
+    Object.defineProperty(compilation, "runtimeBytecodeCborAuxdata", {
+      value: runtimeCodeArtifacts.cborAuxdata,
+    });
+
+    const sourcifyChainMock = {
+      getBytecode: async (address: string) => {
+        return `0x${verifiedContract.onchain_runtime_code.toString("hex")}`;
+      },
+      getTx: async (txHash: string) => {
+        return {
+          blockNumber: verifiedContract.block_number,
+          from: `0x${verifiedContract.deployer.toString("hex")}`,
+        };
+      },
+      getContractCreationBytecodeAndReceipt: async (address: string) => {
+        return {
+          creationBytecode: `0x${verifiedContract.onchain_creation_code.toString("hex")}`,
+          txReceipt: {
+            index: verifiedContract.transaction_index,
+          },
+        };
+      },
+      chainId: verifiedContract.chain_id,
+    };
+
+    const verification = new Verification(
+      compilation,
+      sourcifyChainMock as any,
+      `0x${verifiedContract.address.toString("hex")}`,
+      `0x${verifiedContract.transaction_hash.toString("hex")}`,
+    );
+
+    await verification.verify();
+
+    // Get the verification status
+    const verificationStatus = verification.status;
+    const creationMatch =
+      verificationStatus.creationMatch === "perfect" ||
+      verificationStatus.creationMatch === "partial";
+
+    let creationTransformations = null;
+    let creationValues = null;
+    let creationMetadataMatch = null;
+    if (creationMatch) {
+      creationTransformations = verification.transformations.creation.list
+        ? JSON.stringify(verification.transformations.creation.list)
+        : [];
+      creationValues = verification.transformations.creation.values
+        ? JSON.stringify(verification.transformations.creation.values)
+        : {};
+      creationMetadataMatch = verification.status.creationMatch === "perfect";
+    }
+
+    // Begin transaction for database updates
+    await poolClient.query("BEGIN");
+
+    // 1. Update verified_contracts table using direct query
+    await poolClient.query(
+      `UPDATE verified_contracts SET 
+        creation_match = $1,
+        creation_transformations = $2,
+        creation_values = $3,
+        creation_metadata_match = $4
+      WHERE id = $5`,
+      [
+        creationMatch,
+        creationTransformations,
+        creationValues,
+        creationMetadataMatch,
+        sourcifyMatchId,
+      ],
+    );
+
+    // 2. Update sourcify_matches table using direct query instead of updateSourcifyMatch
+    await poolClient.query(
+      `UPDATE sourcify_matches SET 
+        creation_match = $1
+      WHERE verified_contract_id = $2`,
+      [verificationStatus.creationMatch, sourcifyMatchId],
```

Comment:
here it should be `WHERE id = $2`

Reply:
Amazing catch 🙏 In my test I had those two ids with the same values and I didn't catch it